test_unit/test_sink.cpp

```diff
@@ -27,17 +27,21 @@ TEST(Sink, OneSink) {
 using namespace g3;
    AtomicBoolPtr flag = make_shared < atomic<bool >> (false);
    AtomicIntPtr count = make_shared < atomic<int >> (0);
-   {
-      auto worker = g3::LogWorker::createLogWorker();
-      auto handle = worker->addSink(std2::make_unique<ScopedSetTrue>(flag, count), &ScopedSetTrue::ReceiveMsg);
-      EXPECT_FALSE(flag->load());
-      EXPECT_TRUE(0 == count->load());
-      LogMessagePtr message{std2::make_unique<LogMessage>("test", 0, "test", DEBUG)};
-      message.get()->write().append("this message should trigger an atomic increment at the sink");
-      worker->save(message);
-   }
-   EXPECT_TRUE(flag->load());
+   auto worker = g3::LogWorker::createLogWorker();
+   auto handle = worker->addSink(std2::make_unique<ScopedSetTrue>(flag, count), &ScopedSetTrue::ReceiveMsg);
+   EXPECT_FALSE(flag->load());
```

Comment:
Please leave the test as it was and create new tests instead.
This test verifies that a sink can be added and it can be verified that the sink can receive data. )

From your tests I expect that 
1. A sink is added
(1b. Optionally verify that the sink can receive data. This could be done with a shared data counter)
2. A sink is removed
3. A flag is set to prove that the sink was destroyed
(4. Optionally verify that once the sink is gone sending log messages will not increment the data counter)

It should be fairly basic since a sink can be any construct you choose.